pkg/git/repository.go

```diff
@@ -121,3 +122,38 @@ func (r *Repository) GetLatestCommit() (*object.Commit, error) {
 	}
 	return commit, nil
 }
+
+func (r *Repository) CommitDeployment(DeploymentFileRepositoryPath, KubernetesDeploymentFilePath string) error {
+	// get worktree
+	w, err := r.Worktree()
+	if err != nil {
+		return ErrGeneratingWorkTree{}
```

Comment:
consider augmenting the error with the error that comes from the git package to provide additional information